File: problems/Two Sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const store = new Map();

    for (let i = 0; nums.length > i; ++i) {
        if (store.has(target - nums[i])) {
            return [store.get(target - nums[i]), i];
        } else {
            store.set(nums[i], i);
        }
    }
};

File: problems/Two Sum/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> store = new HashMap<>();

        for (int idx = 0; nums.length > idx; ++idx) {
            if (store.containsKey(target - nums[idx])) {
                return new int[] {store.get(target - nums[idx]), idx};
            }

            store.put(nums[idx], idx);
        }

        return null;
    }
}
